fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -270,7 +277,15 @@ def create_report_thread_length(
     include_all_columns=False
 ) -> pd.DataFrame:
     """Report: thread lengths
+
     include_all_columns: Pending bugfix. If false, excludes these columns from report.
+
+    Warning: If this report is altered to show aggregate results (e.g. message counts, average thread length, standard
```

Comment:
@rohaher This may be something that we have to address. I can do it, though. I just want you to know about it.

I think you may be in the process of adding back summary statistics for the "thread lengths" report aggregated not just at the keyword level, but also at the category and stream levels. When that work is completed, I can fix this issue I'm about to explain if you want. Regardless, I want to explain the issue here and how I fixed it.

---

I noticed a problem, with how we're aggregating; something I mentioned the last time we met with Davera.

The comment that I wrote here says:
> Warning: If this report is altered to show aggregate results (e.g. message counts, average thread length, standard deviation of thread length) at the stream or category level), it is necessary to filter out duplicate threads that are **matched by multiple keywords**. For reference, this is done in the 'user participation' report, identifying threads by the combination of the 'stream' ('stream_id' or 'display_recipient') field and the 'thread name' (subject) field, and then de-duping.

Here's how I solved this problem in my "user participation counts" report:
```py
df_i2 = df_i[df_i['stream'] == s]  # df_i was already filtered by 1 unique user. This part is in a loop and now filtering by unique stream 's'
df_i2_unique_threads = copy(df_i2)  # copying the dataframe so that I can make the changes that I need to get accurate counts, without changing df_i2, which I'll need later in its unaltered state
df_i2_unique_threads['thread.id'] = df_i2_unique_threads['thread.id'].apply(lambda x: str(x))  # I made a synthetic field earlier in the function called 'thread.id', which is a list of [stream_name, thread_name] (just in case someone makes the same thread in multiple streams; I think I've done that before). But before I can filter, I need to change the list to a string
df_i2_unique_threads = df_i2_unique_threads.drop_duplicates(subset='thread.id')  # filtering out by unique thread
role_counts: Dict[str, int] = df_i2_unique_threads['user.role'].value_counts().to_dict()  # counting by from the deduped list of threads
```